### DISCUSSION TITLE
myaccount password issues

### DISCUSSION THREAD

**Discussion by Movix** [eldy/SellYourSaas#363] [Q&A]
Hello,

I encounter a weird behaviour with the myaccount password.
After registration the saas instance is created, i can login there without any issues in the instance itself.
The regsitered customer is automatically logged in myaccount after registration. But after a logout impossible to login again. Credentials are not accepted.
Trying to reset the password in the thirdparty card of the customer within Dolibarr does not work, credentials still invalid on myaccount login.
The only way to get the customer credentials working is to request a password change from the myaccount login page and do the password reset from there.

What am i doing wrong ?

Movix

**Comment by DanWin** [comment]
Looking into it, I found that in the database table `llx_societe_extrafields` the password is stored. After registration it contains the clear text password in a crypted format like this and the password is visible from the 3rd-party module:
`dolcrypt:AES-256-CTR:793c744f3532eb95:5QXazFwRiMu2uDtXaipdWHRQoij5QJm6wAOEnQKFttw=`
After resetting the password, it is changed into BCRYPT format, so the cleartext password can not be retrieved from the 3rd party module anymore, but the login works correctly.
`dolcrypt:AES-256-CTR:9613e7f9dcfd403b:gXoq+ehpfJxKglrpzGFyCQxbLpnhWPpKywB6CT9ZyA3pT4dIhBW0h3nW8RlGq5LNc93WhDAA23drTalH`

It looks like registration and login/password-reset make different assumptions on how to save the password.
